dali/operators/image/remap/warp_affine_params.h

```diff
@@ -109,37 +115,53 @@ class WarpAffineParamProvider
     }
   }
 
-  void UseInputAsParams(const TensorList<CPUBackend> &input) {
+  void UseInputAsParams(const TensorList<CPUBackend> &input, bool invert) {
     CheckParamInput(input);
 
-    params_cpu_.data = static_cast<const MappingParams *>(input.raw_data());
-    params_cpu_.shape = { num_samples_ };
+    if (invert) {
+      auto *params = this->AllocParams(kernels::AllocType::Host);
+      for (int i = 0; i < num_samples_; i++) {
+        if (invert)
+          params[i] = static_cast<const MappingParams *>(input.raw_tensor(i))->invert();
+      }
+    } else {
+      params_cpu_.data = static_cast<const MappingParams *>(input.raw_data());
+      params_cpu_.shape = { num_samples_ };
+    }
   }
 
-  void UseInputAsParams(const TensorVector<GPUBackend> &) {
+  void UseInputAsParams(const TensorVector<GPUBackend> &, bool) {
     DALI_FAIL("This function is here only to avoid excessive complexity of mitigating the call.");
   }
 
-  void UseInputAsParams(const TensorVector<CPUBackend> &input) {
+  void UseInputAsParams(const TensorVector<CPUBackend> &input, bool invert) {
     CheckParamInput(input);
 
-    if (!input.IsContiguous()) {
+    if (!input.IsContiguous() || invert) {
       auto *params = this->AllocParams(kernels::AllocType::Host);
       for (int i = 0; i < num_samples_; i++) {
-        auto &tensor = input[i];
-        params[i] = *static_cast<const MappingParams *>(input[i].raw_data());
+        if (invert)
+          params[i] = static_cast<const MappingParams *>(input[i].raw_data())->invert();
+        else
+          params[i] = *static_cast<const MappingParams *>(input[i].raw_data());
```

Comment:
same here, I don't get how you are supposed to access params later. Maybe I am missing something?